src/Admin/OptionsMenu.php

```diff
@@ -246,6 +255,8 @@ public function enqueue_assets( $hook_suffix ) {
 			'UPDATES_NONCE'                      => wp_create_nonce( 'updates' ),
 			'USER_FIELD_DEVELOPER_TOOLS_ENABLED' => UserAccess::USER_FIELD_DEVELOPER_TOOLS_ENABLED,
 			'USERS_RESOURCE_REST_PATH'           => '/wp/v2/users',
+			'SITE_HAS_CACHE_ENABLE'              => $site_has_cache_enable,
```

Comment:
```suggestion
			'HAS_PAGE_CACHING'              => ( ( defined( 'WP_CACHE' ) && WP_CACHE ) ) || $this->site_health->has_page_caching(),
```

Reply:
Nevertheless, we actually cannot call `has_page_caching` here, as it will do the loopback requests and cause a slowdown. What we need rather is to obtain the _cached_ result. I suggest adding a parameter to `has_page_caching` which allows us to use the previously-computed result.

Here's my suggestion:

```diff
diff --git a/src/Admin/SiteHealth.php b/src/Admin/SiteHealth.php
index 269c8ce64..a149227e5 100644
--- a/src/Admin/SiteHealth.php
+++ b/src/Admin/SiteHealth.php
@@ -246,7 +246,7 @@ private function get_persistent_object_cache_learn_more_action() {
 	 */
 	public function persistent_object_cache() {
 		$is_using_object_cache = wp_using_ext_object_cache();
-		$has_page_caching      = get_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY );
+		$has_page_caching      = $this->has_page_caching( true );
 
 		$description = '<p>' . esc_html__( 'The AMP plugin performs at its best when persistent object cache is enabled. Persistent object caching is used to more effectively store image dimensions and parsed CSS using a caching backend rather than using the options table in the database.', 'amp' ) . '</p>';
 
@@ -337,8 +337,6 @@ static function ( $available_service ) {
 	public function page_cache() {
 		$has_page_caching = $this->has_page_caching();
 
-		set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, true === $has_page_caching, MONTH_IN_SECONDS );
-
 		$badge_color = 'orange';
 		$status      = 'recommended';
 		$label       = __( 'Page caching is not detected', 'amp' );
@@ -385,9 +383,34 @@ public function page_cache() {
 	/**
 	 * Check if site has page cache enable or not.
 	 *
+	 * @param bool $use_previous_result Whether to use the previously-obtained cache result.
 	 * @return bool|WP_Error Whether page caching was detected, or else error information.
 	 */
-	public function has_page_caching() {
+	public function has_page_caching( $use_previous_result = false ) {
+		if ( $use_previous_result ) {
+			$has_page_caching = get_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY );
+			if ( is_wp_error( $has_page_caching ) ) {
+				return $has_page_caching;
+			} elseif ( $has_page_caching ) {
+				return 'yes' === $has_page_caching;
+			}
+		}
+
+		$has_page_caching = $this->check_for_page_caching();
+		if ( is_wp_error( $has_page_caching ) ) {
+			set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $has_page_caching, HOUR_IN_SECONDS );
+		} else {
+			set_transient( self::HAS_PAGE_CACHING_TRANSIENT_KEY, $has_page_caching ? 'yes' : 'no', MONTH_IN_SECONDS );
+		}
+
+		return $has_page_caching;
+	}
+
+	/**
+	 * @return bool|WP_Error
+	 */
+	private function check_for_page_caching() {
+
 		/** This filter is documented in wp-includes/class-wp-http-streams.php */
 		$sslverify = apply_filters( 'https_local_ssl_verify', false );
 
```